### PR TITLE
fix issue #3861: Flood on buildings with no aggregation fails

### PR DESCRIPTION
**Pull request by lucernae**
### What does it fix ?
* Ticket #3861
* Description : handle big numbers in aggregation.

### CODE REVIEW

File: safe/report/extractors/aggregate_result.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def aggregation_result_extractor(impact_report, component_metadata):
         total_affected_value = format_number(
             feat[total_field_index],
             enable_rounding=is_rounded)
-        if total_affected_value == 0:
+        if total_affected_value == '0':
```
Comment (Gustry) [inline safe/report/extractors/aggregate_result.py]: Maybe you should check `total_affected_value in ['0', 0]` ?
Can it happen ?

File: safe/report/extractors/aggregate_result.py [hunk 2]
```diff
@@ -156,7 +156,8 @@ def aggregation_result_extractor(impact_report, component_metadata):
 
     if value_dict:
         for type_name in type_fields:
-            if int(value_dict[type_name]) == 0:
+            affected_value_string_formatted = value_dict[type_name]
+            if affected_value_string_formatted == '0':
```
Comment (Gustry) [inline safe/report/extractors/aggregate_result.py]: Same for here ?

File: safe/report/extractors/aggregate_result.py (hunk 1 above)
Comment (lucernae) [inline safe/report/extractors/aggregate_result.py]: The value now always returned by format_number, so I think it's always a string (with decimal and thousand separator in debug mode)

File: safe/report/extractors/aggregate_result.py (hunk 2 above)
Comment (lucernae) [inline safe/report/extractors/aggregate_result.py]: Same as above, it's now returned from format_number

**Pull request merged**